hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceImpl.java

```diff
@@ -132,9 +132,10 @@ protected HazelcastInstanceImpl(String name, Config config, NodeContext nodeCont
 
         try {
             logger = node.getLogger(getClass().getName());
-            lifecycleService.fireLifecycleEvent(STARTING);
```

Comment:
`Node.start()` is initializing internal stuff. I am not sure if it makes a difference to the user to get the `STARTING` event before internal initialization starts or not. If you think it matters, we can register the listeners first and invoke the lifecycle listener before `start()` is invoked.